### PR TITLE
Install unstable MSIs from artifactory for power shell

### PR DESCRIPTION
**Pull request by patrick-wright**

### CODE REVIEW

File: lib/mixlib/install/generator/powershell/scripts/get_project_metadata_for_artifactory.ps1.erb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  Write-Verbose "Project: $project"
+
+  <% artifacts.each do |artifact| %>
+    $artifact_info_dir = "$($env:temp)/artifact_info/<%= File.join(artifact.platform, artifact.platform_version, artifact.architecture)%>"
+    New-Item -ItemType directory -Path $artifact_info_dir -force
+    New-Item -ItemType file "$($artifact_info_dir)/artifact_info" -force
+    "url <%= artifact.url%>" | out-file "$artifact_info_dir/artifact_info"
+    "md5 <%= artifact.md5%>" | out-file "$artifact_info_dir/artifact_info" -Append
+    "sha256 <%= artifact.sha256%>" | out-file "$artifact_info_dir/artifact_info" -Append
+    "version <%= artifact.version%>" | out-file "$artifact_info_dir/artifact_info" -Append
```
Comment (sersut) [inline lib/mixlib/install/generator/powershell/scripts/get_project_metadata_for_artifactory.ps1.erb]: We do not need to write and read the version on this artifact. We already have the version available to us as `$version` at this point in the script.

File: lib/mixlib/install/generator/powershell/scripts/get_project_metadata_for_artifactory.ps1.erb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  <% end %>
+
+  $artifact_info_for_platform = Get-Content "$($env:temp)/artifact_info/$($platform)/$($platform_version)/$($machine)/artifact_info"
+
+  $package_metadata = ($artifact_info_for_platform).trim() -split '\n' |
+    foreach { $hash = @{} } {$key, $value = $_ -split '\s+'; $hash.Add($key, $value)} {$hash}
```
Comment (sersut) [inline lib/mixlib/install/generator/powershell/scripts/get_project_metadata_for_artifactory.ps1.erb]: Nice :smile:

**Comment by sersut** [comment]
Looks good @patrick-wright. A few minor comments. Also would be nice to squash the commits and have a good commit message.

File: lib/mixlib/install/generator/bourne/scripts/artifactory_urls.sh.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+############
+
+<% artifacts.each do |artifact| %>
+  artifact_info_dir="$tmp_dir/artifact_info/<%= File.join(artifact.platform, artifact.platform_version, artifact.architecture)%>"
+  mkdir -p $artifact_info_dir
+  touch "$artifact_info_dir/artifact_info"
+  echo "url <%= artifact.url%>" >> "$artifact_info_dir/artifact_info"
+  echo "md5 <%= artifact.md5%>" >> "$artifact_info_dir/artifact_info"
+  echo "sha256 <%= artifact.sha256%>" >> "$artifact_info_dir/artifact_info"
+  echo "version <%= artifact.version%>" >> "$artifact_info_dir/artifact_info"
```
Comment (patrick-wright) [inline lib/mixlib/install/generator/bourne/scripts/artifactory_urls.sh.erb]: @sersut I replicated this logic.  Is version needed here?

**Comment by patrick-wright** [comment]
@sersut Should be good to go now.  See my comment above about `artifactory_urls.sh`.

File: lib/mixlib/install/generator/bourne/scripts/artifactory_urls.sh.erb (hunk 3 above)
Comment (sersut) [inline lib/mixlib/install/generator/bourne/scripts/artifactory_urls.sh.erb]: Nice catch. Version is not needed here. I will remove it on your branch.

**Pull request merged**